### ISSUE TITLE
(Android Only) New Architecture Support Required

### ISSUE THREAD

**Issue by MoamberRaza** [react-native-checkbox/react-native-checkbox#229]
Not supporting New Architecture support currently available due to which android tap is a reason of crashing app.

**Comment by MoamberRaza** [comment]
Any Update.

**Comment by Satyam-code143** [comment]
I am facing the same issue as well. Any update on this?

**Comment by shashi-patel** [comment]
I am also facing the same issue. Please let me know if you have any updates on this.

**Comment by saeid-elio** [comment]
Is there any alternative to this package that one can use with latest version of React Native (0.78) as this package is crashing my app on android phones.

**Comment by siddharthsprybe** [comment]
I'm facing the same issue too, is there an alternative package ?

**Comment by AndreAparecidoDaSilva** [comment]
I managed to work around this issue by placing the CheckBox inside a TouchableOpacity that actually changes the state and setting the CheckBox's disabled prop to {true}.
It would be something like this:

const [myState, setMyState] = useState(false);

        <TouchableOpacity
            style={styles.check}
            onPress={() => setMyState(!myState)}
        >
                <CheckBox
                    disabled={true}
                    value={myState}
                    onValueChange={(newValue) => setMyState(newValue)}
                    tintColors={{ true: '#516091', false: '#516091' }}
                />

        </TouchableOpacity>

**Comment by shashi-patel** [comment]
You can use the following alternative package. I upgraded to this package to resolve the issue and it works well:

https://www.npmjs.com/package/react-native-bouncy-checkbox

**Comment by saeid-elio** [comment]
Yes, I can second @shashi-patel's suggestion. React-native-bouncy-checkbox not only works on both ios and android in latest version of React Native, but also provides a more consistent UI on both platforms.

**Comment by MoamberRaza** [comment]
Appreciated, but i cannot suddenly go for new lib or new submission. until is there any solution for this. little constraints 😑

**Comment by dsolimando** [comment]
PR with a fix [here](https://github.com/react-native-checkbox/react-native-checkbox/pull/231)

**Comment by nicholaslee119** [comment]
Thanks for your PR, it is release as 0.5.20

**Comment by Gonky10** [comment]
Hi, is the problem fixed? in react native 0.81

**Comment by dave-mp** [comment]
expo-doctor still says with version 0.5.20:

`Unsupported on New Architecture: @react-native-community/checkbox`

**Comment by nicholaslee119** [comment]
The Android new-architecture crash (NPE on tap) was fixed in #231 and the codebase now builds against the New Architecture (Android `IS_NEW_ARCHITECTURE_ENABLED`, codegen config, Windows on new arch via #239). Closing the Android crash issue as fixed. Note: on iOS this component runs through the new-arch interop layer rather than as a native Fabric component — if you rely on `expo-doctor` reporting full Fabric support, follow #211. Please reopen if the Android crash still reproduces on the latest version.
